mm2src/lp_ordermatch.rs

```diff
@@ -2807,43 +2793,67 @@ pub async fn lp_ordermatch_loop(ctx: MmArc) {
             let mut my_maker_orders = ordermatch_ctx.my_maker_orders.lock().await;
             let mut to_cancel = vec![];
 
-            for (uuid, order) in my_maker_orders.iter() {
+            for (uuid, order) in my_maker_orders.iter_mut() {
                 if !ordermatch_ctx.orderbook.lock().await.order_set.contains_key(uuid) {
-                    if let Ok(Some((base, rel))) = find_pair(&ctx, &order.base, &order.rel).await {
-                        if let Err(e) = order.check_balance(&ctx, &base, &rel).await {
+                    let (base, rel) = match find_pair(&ctx, &order.base, &order.rel).await {
+                        Ok(Some(coins)) => coins,
+                        _ => continue,
+                    };
+
+                    let current_balance = match base.my_spendable_balance().compat().await {
+                        Ok(b) => b,
+                        Err(e) => {
                             log::info!("Error {} on balance check to kickstart order {}, cancelling", e, uuid);
                             to_cancel.push(*uuid);
                             continue;
-                        }
+                        },
+                    };
+                    let max_vol =
+                        match calc_max_maker_vol(&ctx, &base, &current_balance, FeeApproxStage::OrderIssue).await {
```

Comment:
Can we avoid calculating max maker volume somehow on every `lp_ordermatch_loop` iteration?
This function requires significant network usage.

Reply:
It's not called on every iteration actually. It's called only for our orders, which exist in `my_maker_orders` but do not exist in `order_set` yet. It happens only for kickstarted orders. If we don't calculate it, we may send the wrong order volume to other peers.

Reply:
Thanks! I missed this fact!
Maybe we can move this logic to a separated function like `kickstart_maker_order` or `kickstart_missing_order`?

Reply:
We can, but we need this PR merged if there are no major flaws. Because it's blocking the release.